### PR TITLE
GCE- Break Terraform configuration into modules

### PR DESCRIPTION
**Pull request by anners**
Broke the original gce.tf file into 4 modules:
network
nodes/control
nodes/edge
nodes/worker

added support for multi-zones
added examples on how to use remote state with terraform
updated the docs and gce.sample.tf to reflect all the changes

### CODE REVIEW

**Comment by siddharthist** [comment]
Awesome, thank you! We'll review soon.

**Comment by siddharthist** [comment]
@anners Unfortunately, some other changes have created merge conflicts with this branch. Would you mind rebasing this work onto current master?

**Comment by anners** [comment]
@siddharthist I think I have fixed the issues.

File: terraform/gce/nodes/worker/main.tf [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+variable "long_name" {}
+variable "network_name" {}
+variable "short_name" {}
+variable "ssh_key" {}
+variable "ssh_user" {}
+variable "zones" {}
+
+
+# Instances
+resource "google_compute_disk" "manlt-worker-lvm" {
```
Comment (sehqlr) [inline terraform/gce/nodes/worker/main.tf]: There's a typo here. It should be `mantl`, not `manlt`

File: terraform/gce/nodes/worker/main.tf [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  name = "${var.short_name}-worker-lvm-${format("%02d", count.index+1)}"
+  type = "pd-ssd"
+  zone = "${element(split(",", var.zones), count.index)}"
+  size = "${var.worker_data_volume_size}"
+
+  count = "${var.worker_count}"
+}
+
+
+resource "google_compute_instance" "manlt-worker-nodes" {
```
Comment (sehqlr) [inline terraform/gce/nodes/worker/main.tf]: same typo here

**Comment by anners** [comment]
Thanks @sehqlr for catching my bad search and replace. :)

**Comment by ryane** [comment]
fyi, gce released a centos 7.2 image a few days ago. see #1015

**Comment by anners** [comment]
I have updated image to centos-7-v20160119.

**Comment by siddharthist** [comment]
@ryane Does this look ready for testing?

**Comment by ryane** [comment]
@siddharthist I am not sure - I think the terraform refactorings are still in flux. but @anners and @stevendborrelli would know better.

**Comment by anners** [comment]
@stevendborrelli thoughts on merging this, since the similar AWS branch has been merged.

**Comment by sehqlr** [comment]
@anners Sorry it took so long, but I was able to test your changes. I've confirmed that they work. The merge conflict is in your version of `.gitignore`. Unless there is anything else that you need to change, I can fix the conflict and merge this.

**Comment by anners** [comment]
go for it! Thanks.

**Pull request merged**